devito/types/dimension.py

```diff
@@ -607,6 +607,65 @@ def middle(cls, name, parent, thickness_left, thickness_right, local=False):
                    thickness=((lst, thickness_left), (rst, thickness_right)),
                    local=local)
 
+    @classmethod
+    def intersection(cls, sdims):
+        # Check all SubDimensions supplied overlap
+        # NOTE: Think the test for overlap isn't strict enough?
+        if not all([sdims[0].overlap(sdim) for sdim in sdims[1:]]):
+            raise ValueError("SubDimensions do not all overlap")
+
+        name = '_n_'.join([sdim.name for sdim in sdims])
```

Comment:
`_i_` ?

Reply:
Was thinking it looks like the set intersection symbol if you squint at it?